pallets/inflation/src/tests.rs

```diff
@@ -0,0 +1,473 @@
+}
+
+#[test]
+fn force_set_inflation_params_work() {
+    ExternalityBuilder::build().execute_with(|| {
+        let mut new_params = InflationParams::<Test>::get();
+        new_params.max_inflation_rate = Perquintill::from_percent(20);
+        assert!(new_params != InflationParams::<Test>::get(), "Sanity check");
+
+        // Execute call, ensure it works
+        assert_ok!(Inflation::force_set_inflation_params(
+            RuntimeOrigin::root(),
+            new_params
+        ));
+        System::assert_last_event(Event::InflationParametersForceChanged.into());
+
+        assert_eq!(InflationParams::<Test>::get(), new_params);
+    })
+}
+
+#[test]
+fn force_set_inflation_params_fails() {
+    ExternalityBuilder::build().execute_with(|| {
+        let mut new_params = InflationParams::<Test>::get();
+        new_params.base_stakers_part = Zero::zero();
+        assert!(
+            !new_params.is_valid(),
+            "Must be invalid for check to make sense."
+        );
+
+        // Make sure it's not possible to force-set invalid params
+        assert_noop!(
+            Inflation::force_set_inflation_params(RuntimeOrigin::root(), new_params),
+            Error::<Test>::InvalidInflationParameters
+        );
+
+        // Make sure action is privileged
+        assert_noop!(
+            Inflation::force_set_inflation_params(RuntimeOrigin::signed(1).into(), new_params,),
+            BadOrigin
+        );
+    })
+}
+
+#[test]
+fn force_set_inflation_config_work() {
+    ExternalityBuilder::build().execute_with(|| {
+        let mut new_config = ActiveInflationConfig::<Test>::get();
+        new_config.recalculation_block = new_config.recalculation_block + 50;
+
+        // Execute call, ensure it works
+        assert_ok!(Inflation::force_set_inflation_config(
+            RuntimeOrigin::root(),
+            new_config
+        ));
+        System::assert_last_event(
+            Event::InflationConfigurationForceChanged { config: new_config }.into(),
+        );
+
+        assert_eq!(ActiveInflationConfig::<Test>::get(), new_config);
+    })
+}
+
+#[test]
+fn force_set_inflation_config_fails() {
+    ExternalityBuilder::build().execute_with(|| {
+        let mut new_config = ActiveInflationConfig::<Test>::get();
+        new_config.recalculation_block = new_config.recalculation_block + 50;
+
+        // Make sure action is privileged
+        assert_noop!(
+            Inflation::force_set_inflation_config(RuntimeOrigin::signed(1), new_config),
+            BadOrigin
+        );
+    })
+}
+
+#[test]
+fn force_inflation_recalculation_work() {
+    ExternalityBuilder::build().execute_with(|| {
+        let old_config = ActiveInflationConfig::<Test>::get();
+
+        // Execute call, ensure it works
+        assert_ok!(Inflation::force_inflation_recalculation(
+            RuntimeOrigin::root(),
+        ));
+
+        let new_config = ActiveInflationConfig::<Test>::get();
+        assert!(
```

Comment:
assert_ne!(
            old_config, new_config,
            "Config should change, otherwise test doesn't make sense."
        );

Reply:
Tbh, I didn't even know this existed 😂